applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_shell_element.cpp

```diff
@@ -0,0 +1,390 @@
+            }
+            case TracedStressType::FZZ:
+            {
+                direction_1 = 2;
+                direction_2 = 2;
+                stress_is_moment = false;
+                break;
+            }
+            default:
+                KRATOS_ERROR << "Invalid stress type! Stress type not supported for this element!" << std::endl;
+        }
+
+        if(stress_is_moment)
+            mpPrimalElement->GetValueOnIntegrationPoints(SHELL_MOMENT_GLOBAL, stress_vector, rCurrentProcessInfo);
+        else
+            mpPrimalElement->GetValueOnIntegrationPoints(SHELL_FORCE_GLOBAL, stress_vector, rCurrentProcessInfo);
+
+        rOutput.resize(num_gps);
+        for(size_t i = 0; i < num_gps; i++)
+        {
+            rOutput(i) = stress_vector[i](direction_1, direction_2);
+        }
+
+    }
+    else
+    {
+        rOutput.resize(num_gps);
+        rOutput.clear();
+    }
+
+    KRATOS_CATCH("")
+}
+
+int AdjointFiniteDifferencingShellElement::Check(const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+
+    GeometryType& r_geom = GetGeometry();
+
+    // verify that the variables are correctly initialized
+    KRATOS_CHECK_VARIABLE_KEY(DISPLACEMENT);
+    KRATOS_CHECK_VARIABLE_KEY(ROTATION);
+    KRATOS_CHECK_VARIABLE_KEY(VELOCITY);
+    KRATOS_CHECK_VARIABLE_KEY(ACCELERATION);
+    KRATOS_CHECK_VARIABLE_KEY(DENSITY);
+    KRATOS_CHECK_VARIABLE_KEY(SHELL_CROSS_SECTION);
+    KRATOS_CHECK_VARIABLE_KEY(THICKNESS);
+    KRATOS_CHECK_VARIABLE_KEY(CONSTITUTIVE_LAW);
+    KRATOS_CHECK_VARIABLE_KEY(ADJOINT_DISPLACEMENT);
+    KRATOS_CHECK_VARIABLE_KEY(ADJOINT_ROTATION);
+
+    // check properties
+    KRATOS_ERROR_IF(this->pGetProperties() == nullptr) << "Properties not provided for element " << this->Id() << std::endl;
+
+    const PropertiesType & props = this->GetProperties();
+
+    if(props.Has(SHELL_CROSS_SECTION)) // if the user specified a cross section ...
+    {
+        const ShellCrossSection::Pointer & section = props[SHELL_CROSS_SECTION];
+        KRATOS_ERROR_IF(section == nullptr) << "SHELL_CROSS_SECTION not provided for element " << this->Id() << std::endl;
+
+        section->Check(props, r_geom, rCurrentProcessInfo);
+    }
+    else // ... allow the automatic creation of a homogeneous section from a material and a thickness
+    {
+        KRATOS_ERROR_IF_NOT(props.Has(CONSTITUTIVE_LAW)) << "CONSTITUTIVE_LAW not provided for element " << this->Id() << std::endl;
+        const ConstitutiveLaw::Pointer& claw = props[CONSTITUTIVE_LAW];
+        KRATOS_ERROR_IF(claw == nullptr) << "CONSTITUTIVE_LAW not provided for element " << this->Id() << std::endl;
+
+        KRATOS_ERROR_IF_NOT(props.Has(THICKNESS)) <<  "THICKNESS not provided for element " <<  this->Id() << std::endl;
+        KRATOS_ERROR_IF(props[THICKNESS] <= 0.0) << "wrong THICKNESS value provided for element " << this->Id() << std::endl;
+
+        ShellCrossSection::Pointer dummySection = ShellCrossSection::Pointer(new ShellCrossSection());
```

Comment:
make shared

Reply:
Done, however it is also like this in the base shell element https://github.com/KratosMultiphysics/Kratos/blob/f915b9091360d340a48109912506b6e401f5e587/applications/StructuralMechanicsApplication/custom_elements/base_shell_element.cpp#L504